device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -636,7 +636,15 @@ public void onDeviceSessionOpened(String deviceId)
         {
             log.trace("Device session for device {} opened, counting down the device sessions opening latch", deviceId);
             this.deviceSessionsOpenedLatches.get(deviceId).countDown();
-            this.listener.onMultiplexedDeviceSessionEstablished(this.connectionId, deviceId);
+
+            if (isMultiplexing)
+            {
+                // For non-multiplexing cases, the above logic for counting down the latch for the single device session
+                // will notify the open() thread that the reactor thread has successfully opened the device session.
+                // That will then execute the onConnectionEstablished callback on this same listener. Because of that,
+                // there is no listener callback to execute here for non-multiplexing cases.
+                this.listener.onMultiplexedDeviceSessionEstablished(this.connectionId, deviceId);
+            }
```

Comment:
I found this bug while investigating this issue. We shouldn't have been calling "on**Multiplexed**DeviceSessionEstablished" when not multiplexing. It resulted in the SDK making multiple internal connection state updates when only one should have happened.

Reply:
This bug shouldn't have any meaningful impact on a user, but it did make debugging the connection state reporting confusing